In [8]:
import sys
sys.path.append("/home/ujjwal/codes/building_api")

from app.models import League, Player, Team, Performance, TeamPlayer
from database import Base, engine, SessionLocal

In [11]:
db= SessionLocal() 
players = db.query(Player).all()
len(players)

1018

In [18]:
first_player= db.query(Player).first()
first_player.first_name

'Aaron'

In [ ]:
## get player name 
id= 1001
db.query(Player).filter(Player.player_id==id).first().first_name


'Aaron'

In [74]:
## get players  by  min_last_changed
min_last_changed = '2024-02-20 00:00:00'
skip = 100
limit= 50
query= db.query(Player).filter(Player.last_changed_date >=min_last_changed)

answers= query.offset(skip).limit(limit).all()

for answer in answers:
    print(answer.player_id, answer.first_name, answer.last_name, answer.last_changed_date)

1101 Dwayne Washington 2024-04-18
1102 Kalif Raymond 2024-04-18
1103 Alex Erickson 2024-04-18
1104 Wil Lutz 2024-04-18
1105 Robbie Chosen 2024-04-18
1106 Stephen Anderson 2024-04-18
1107 Ka'imi Fairbairn 2024-04-18
1108 Derrick Henry 2024-04-18
1109 Michael Thomas 2024-04-18
1110 Demarcus Robinson 2024-04-18
1111 Jakeem Grant 2024-04-18
1112 Nate Sudfeld 2024-04-18
1113 C.J. Ham 2024-04-18
1114 Carson Wentz 2024-04-18
1115 Laquon Treadwell 2024-04-18
1116 Jonathan Williams 2024-04-18
1117 J.P. Holtz 2024-04-18
1118 Jaydon Mickens 2024-04-18
1119 Tyler Boyd 2024-04-18
1120 Tyreek Hill 2024-04-18
1121 Ezekiel Elliott 2024-04-18
1122 Dak Prescott 2024-04-18
1123 Hunter Henry 2024-04-18
1124 Jared Goff 2024-04-18
1125 Tyler Higbee 2024-04-18
1126 Kenyan Drake 2024-04-18
1127 Jacoby Brissett 2024-04-18
1128 Mo Alie-Cox 2024-04-18
1129 Johnny Mundt 2024-04-18
1130 Zach Pascal 2024-04-18
1131 Darrell Daniels 2024-04-18
1132 PJ Walker 2024-04-18
1133 Christian McCaffrey 2024-04-18
1134 Curtis 

In [82]:
from sqlalchemy.orm import Session, joinedload
from datetime import date

In [91]:
min_last_changed_date = date(2024, 4, 18)

query = query.filter(models.Player.last_changed_date >= min_last_changed_date)
query

answers= query.offset(20).limit(10).all()
for answer in answers:
    print(answer.player_id, answer.first_name, answer.last_name, answer.last_changed_date)

1021 Cole Beasley 2024-04-18
1022 Brandon Bolden 2024-04-18
1023 Russell Wilson 2024-04-18
1024 Marvin Jones 2024-04-18
1025 Randy Bullock 2024-04-18
1026 Justin Tucker 2024-04-18
1027 Kirk Cousins 2024-04-18
1028 Greg Zuerlein 2024-04-18
1029 Ryan Tannehill 2024-04-18
1030 Brandon McManus 2024-04-18


## due to offset 
offset = 20 so 
, it starts from 1021 insited of 1000 
and 
limit is 10 so only show 10 results